eng/templates/variables-template.yml

```diff
@@ -0,0 +1,20 @@
+# variables-template.yml
+parameters:
+- name: buildConfig
+  default: 'Release'
+
+variables:
+  - name: _TeamName
+    value: DotNetTry
+  - name: _BuildConfig
+    value: ${{ parameters.buildConfig }}
```

Comment:
I am not sure this will work the way you expect it to.  I believe you will have to specify _BuildConfig at the top level.  Variables are resolved at compile time, not run time.

Reply:
Yes, you are right that this gets resolved at compile time.

The values for _BuildConfig is already hardcoded. In azure-pipelines.yml the value is Debug and in azure-pipelines-CI.yml the value is Release and in the pipeline execution it is taking the correct configuration. It seems that the preprocessor already figures out the hardcoded value.

Reply:
I set to Release for both pipelines and that removed the parameter.buildConfig.

done